graph/task.go

```diff
@@ -251,7 +251,9 @@ func (t *Task) initialize(ctx context.Context) error {
 		s.StepStatus = Skipped
 
 		if s.IsBuildStep() {
-			s.Tags = util.ParseTags(s.Build)
+			if s.Tags != nil && len(s.Tags) == 0 {
```

Comment:
Why did you change this?

Reply:
Tha's because in the `build.go` we grab the tags and prefix them with registry (if necessary). I also added the list of those prefixedTags to the buildStep so that we don;'t have to go thru the command again.

```
// After the template has rendered, we have to parse the tags again
	// so we can properly set the build/push tags.
	rendered, prefixedTags := util.PrefixTags(rendered, registry)
	tags = prefixedTags

	buildStep := &graph.Step{
		ID:          "build",
		Build:       rendered,
		Timeout:     buildTimeoutInSec,
		EnableCache: enableCache,
		CacheID:     cacheID,
		Tags:        tags,    // <----
	}
```

Reply:
and what about for exec

Reply:
Sorry, there was a bug in my code earlier. I fixed the condition now.

basically if the `s.Tags` is not set already, parseTags. 
When it goes thru build, they are already set - so we dont parseTags again.
When it goes thru exec, they are not - and we parseTags.